tests/workflow_management.test.ts

```diff
@@ -704,4 +768,106 @@ describe('test-list-steps', () => {
+  });
+
+  test('test-call-child-workflow-last', async () => {
+    const wfid = uuidv4();
+    const handle = await DBOS.startWorkflow(TestListSteps, { workflowID: wfid }).callChildWorkflowLast();
+    await handle.getStatus();
+    await handle.getResult();
+    const wfsteps = await listWorkflowSteps(config, wfid);
+    console.log(wfsteps);
+    expect(wfsteps.steps.length).toBe(4);
+    expect(wfsteps.steps[0].function_name).toBe('stepOne');
+    expect(wfsteps.steps[1].function_name).toBe('stepTwo');
+    expect(wfsteps.steps[2].function_name).toBe('testWorkflow');
+    expect(wfsteps.steps[3].function_name).toBe('getStatus');
+  });
+
+  test('test-queue-child-workflow-first', async () => {
+    const wfid = uuidv4();
+    const handle = await DBOS.startWorkflow(TestListSteps, { workflowID: wfid }).enqueueChildWorkflowFirst();
+    await handle.getStatus();
+    await handle.getResult();
+    const wfsteps = await listWorkflowSteps(config, wfid);
+    console.log(wfsteps);
+    expect(wfsteps.steps.length).toBe(4);
+    expect(wfsteps.steps[0].function_name).toBe('testWorkflow');
+    expect(wfsteps.steps[1].function_name).toBe('getStatus');
+    expect(wfsteps.steps[2].function_name).toBe('stepOne');
+    expect(wfsteps.steps[3].function_name).toBe('stepTwo');
+  });
+
+  test('test-queue-child-workflow-middle', async () => {
+    const wfid = uuidv4();
+    const handle = await DBOS.startWorkflow(TestListSteps, { workflowID: wfid }).enqueueChildWorkflowMiddle();
+    await handle.getStatus();
+    await handle.getResult();
+    const wfsteps = await listWorkflowSteps(config, wfid);
+    console.log(wfsteps);
+    expect(wfsteps.steps.length).toBe(4);
+    expect(wfsteps.steps[0].function_name).toBe('stepOne');
+    expect(wfsteps.steps[1].function_name).toBe('testWorkflow');
+    expect(wfsteps.steps[2].function_name).toBe('getStatus');
+    expect(wfsteps.steps[3].function_name).toBe('stepTwo');
+  });
+
+  test('test-queue-child-workflow-last', async () => {
+    const wfid = uuidv4();
+    const handle = await DBOS.startWorkflow(TestListSteps, { workflowID: wfid }).enqueueChildWorkflowLast();
+    await handle.getStatus();
+    await handle.getResult();
+    const wfsteps = await listWorkflowSteps(config, wfid);
+    console.log(wfsteps);
+    expect(wfsteps.steps.length).toBe(4);
+    expect(wfsteps.steps[0].function_name).toBe('stepOne');
+    expect(wfsteps.steps[1].function_name).toBe('stepTwo');
+    expect(wfsteps.steps[2].function_name).toBe('testWorkflow');
+    expect(wfsteps.steps[3].function_name).toBe('getStatus');
+  });
+
+  test('test-child-rerun', async () => {
+    const wfid = uuidv4();
+    let handle = await DBOS.startWorkflow(TestListSteps, { workflowID: wfid }).CounterParent();
+    await handle.getResult();
+    // call again with same wfid
+    handle = await DBOS.startWorkflow(TestListSteps, { workflowID: wfid }).CounterParent();
+    await handle.getResult();
+
+    expect(callcount).toBe(1);
```

Comment:
The problem with the counter is that it could be flaky based on timing. If the workflow result has been recorded in the database, then DBOS would directly return without incrementing the counter. If the result is not flushed to the database, the counter will always increment.